### DISCUSSION TITLE
Does zotero-better-bibtex can get title in sentence case and ignore proper nouns?

### DISCUSSION THREAD

**Discussion by YoushanLiu** [retorquere/zotero-better-bibtex#2101] [General]
Although the zotero-better-bibtex can get title in BBT sentence case, it still get some proper nouns in lowercase. In the EndNote, there is Preference that is called 'Change Case', which can ignore some proper nouns given by user. Does zotero-better-bibtex can get title in sentence case and ignore proper nouns? I noticed that there is a preferebce option called extensions.zotero.translors.better-bibtex.skipWords. However, I'm not sure its function. Thanks a lot!

**Comment by retorquere** [comment]
skipWords is only for the citekey. If this is for titles, Zotero has a syntax that BBT supports; you can enclose proper nouns in `<span class="nocase">....</span>`. It's not pretty, but it works.

**Reply by YoushanLiu** [reply]
Hi， Heyns. Thank you for your reply!

  Yes, I noticed this function "BBT sentence case". However, it needs edit title one by one. It may not be able to batch convert.

  Could this feature can be included in the future versions by defining some skipWords (using regexp) for title sentence case.

  Thus, some proper nouns (such as United States, America, China, Africa, Tibet, etc.) can be ignored by the BBT sentence case.

  Thanks a lot!

**Reply by retorquere** [reply]
That's a possibility I will consider. But with this same approach I could also get you a javascript for Zotero's javascript runner that would apply this change to all items (eg actually change titles to have the `nocase` code. That way, any changes that are undesirable are under your control, and you'd get the same behavior with Zotero-rendered references and BibTeX-rendered references from the same item.

**Reply by YoushanLiu** [reply]
Hi Heyns. Thank you for your consideration. Where the javarscript can be avaliable? Thanks a lot! Best Wishes.

**Reply by retorquere** [reply]
In Zotero: Menu `Tools`, `Developer`, `Run javascript`, check the checkbox `run as async`, and paste the following under `Code`:

```
for (const item of Zotero.getActiveZoteroPane().getSelectedItems()) {
  const title = item.getField('title')
  const nocase = title.replace(/\b(LHC|NASA)\b/g, '<span class="nocase">$1</span>')
  if (title !== nocase) {
      await item.setField('title', nocase)
      await item.saveTx()
  }
}
```

now select some items in Zotero, and click the `Run` button. change the regex above as needed.

**Reply by YoushanLiu** [reply]
Hi Heyns! Ok! Thank you very much! Best Wishes!

**Reply by retorquere** [reply]
please do test with just one or two items before using it on the lot, as this will change items without asking once you run.

**Reply by YoushanLiu** [reply]
Ok! Thank you for your suggestions! I will try. Thanks a lot!
